### PR TITLE
Reduce Linux build warnings

### PR DESCRIPTION
**Pull request by hacst**
This PR significantly reduces the amount of warnings emitted during a Linux build with gcc 4.9. See the separate commits for more specific information.

To cleanly remove the leftover warnings will require bigger refactoring or semantic changes and will be performed in follow-up. The two big left-over areas is the interface to OpenGL for the overlay where we have a lot of conversion warnings left as well as a deprecation warning in the shortcut code which has to be more closely investigated.

### CODE REVIEW

File: src/murmur/murmur.pro [hunk 1]
```diff
@@ -68,7 +68,7 @@ macx {
 
 dbus {
 	DEFINES *= USE_DBUS
-	CONFIG *= qdbus
+	QT *= dbus
```
Comment (mkrautz) [inline src/murmur/murmur.pro]: Breaks Qt 4, I believe?

File: src/mumble/mumble.pro [hunk 2]
```diff
@@ -525,7 +525,7 @@ bonjour {
 
 dbus {
 	DEFINES *= USE_DBUS
-	CONFIG *= qdbus
+	QT *= dbus
```
Comment (mkrautz) [inline src/mumble/mumble.pro]: Breaks Qt 4, I believe?

File: src/mumble/PulseAudio.cpp [hunk 3]
```diff
@@ -775,7 +775,7 @@ void PulseAudioSystem::stream_restore_read_callback(pa_context *c, const pa_ext_
 	}
 }
 
-void PulseAudioSystem::restore_volume_success_callback(pa_context *c, int success, void *userdata) {
+void PulseAudioSystem::restore_volume_success_callback(pa_context * /*c*/, int /*success*/, void *userdata) {
```
Comment (mkrautz) [inline src/mumble/PulseAudio.cpp]: I hate this coding style.

File: src/mumble/MainWindow.cpp [hunk 4]
```diff
@@ -2449,7 +2449,7 @@ void MainWindow::removeTarget(ShortcutTarget *st)
 		qmCurrentTargets[*st] -= 1;
 }
 
-void MainWindow::on_gsCycleTransmitMode_triggered(bool down, QVariant scdata) 
+void MainWindow::on_gsCycleTransmitMode_triggered(bool down, QVariant /*scdata*/)
```
Comment (mkrautz) [inline src/mumble/MainWindow.cpp]: I hate this coding style.

File: src/mumble/Log.ui [hunk 5]
```diff
@@ -176,7 +176,7 @@
      <property name="title">
       <string>Chat Log</string>
      </property>
-     <layout class="QGridLayout" name="gridLayout">
+     <layout class="QGridLayout" name="_2">
```
Comment (mkrautz) [inline src/mumble/Log.ui]: Unrelated?

File: src/mumble/Log.ui [hunk 6]
```diff
@@ -6,8 +6,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>446</width>
-    <height>334</height>
+    <width>524</width>
```
Comment (mkrautz) [inline src/mumble/Log.ui]: Unrelated?

File: src/murmur/murmur.pro (hunk 1 above)
Comment (hacst) [inline src/murmur/murmur.pro]: Shouldn't see comment. If it does the Qt docs are misleading.

File: src/mumble/PulseAudio.cpp (hunk 3 above)
Comment (hacst) [inline src/mumble/PulseAudio.cpp]: Remove them then? I usually prefer commenting unused params so I don't have to look them up if I end up using them after all or just want to know what other params are available on the callback.

Comment (mkrautz) [inline src/mumble/PulseAudio.cpp]: I myself prefer Q_UNUSED.

File: src/mumble/Log.ui (hunk 5 above)
Comment (hacst) [inline src/mumble/Log.ui]: Nope. Warning emitted because Qt creates "gridLayout" as the default name of the first grid layout in that file which collides with the old value. I just hit reset in the designed and it reset it to "_2". Not sure if I could completely remove it and didn't consider it worth questioning.

File: src/mumble/Log.ui (hunk 6 above)
Comment (hacst) [inline src/mumble/Log.ui]: Yeah. That was qtcreator's doing when I saved it. I remove that if you want. It was buried under other commits and I was lazy....

File: src/mumble/PulseAudio.cpp (hunk 3 above)
Comment (hacst) [inline src/mumble/PulseAudio.cpp]: Mew. That's a way I don't like. Why add extra lines into your function body for a variable you won't use in the whole function. Commenting it out (or removing the name) makes it obvious it is actually unused. Q_UNUSED just tells me that it maybe was at some point unused and could possibly still be.

**Comment by hacst** [comment]
What's left to do for this one? Remove the height/width noise in Log.ui. Anything else? If you really strongly disagree with the commenting out of unused argument names we should discuss that on IRC I guess.

**Comment by mkrautz** [comment]
I simply haven't reviewed thoroughly enough yet... Sorry.

The first round was just a cursory glance. I'd like to make sure we don't break something in landing this.

So, rest assured, I'm not silent because I am mad about the coding style. :-)

**Comment by hacst** [comment]
Sorry. I assumed you were waiting for some changes/comments from me. I sometimes miss a reply and then a PR lingers for ages until I realize I already had the answer I was supposedly waiting for. Just wanted to make sure that wasn't the case.

**Comment by mkrautz** [comment]
Will review asap.

**Comment by mkrautz** [comment]
LGTM with my comments addressed. Sorry for the delay.

**Comment by mkrautz** [comment]
Regarding the unused params: I think as a compromise we should simply drop the parameter names from the new methods with unused parameters.

File: src/mumble/mumble.pro (hunk 2 above)
Comment (hacst) [inline src/mumble/mumble.pro]: See above

**Comment by hacst** [comment]
Tried to address your concerns. Unfortunately I noticed that the way I tried to suppress protobuf warnings was faulty and I had to go a different way (see 6b5c0d7). Seems to work for Linux. I have yet to try if I broke anything on windows and the pro file I wrote for this should be carefully reviewed.

**Comment by mkrautz** [comment]
LGTM.

**Comment by hacst** [comment]
Squashed the fixups into the corresponding commits and did some reordering. Merged as 46cb8a3...4fe1664
